api/src/pcapi/routes/backoffice/non_payment_notices/forms.py

```diff
@@ -0,0 +1,76 @@
+from flask import flash
+from flask_wtf import FlaskForm
+from wtforms.validators import DataRequired
+
+from pcapi.core.offerers import models as offerers_models
+from pcapi.models import db
+from pcapi.routes.backoffice import filters
+from pcapi.routes.backoffice.forms import fields
+from pcapi.routes.backoffice.forms import utils as forms_utils
+
+
+class CreateNonPaymentNoticeForm(FlaskForm):
+    class Meta:
+        locales = ["fr_FR", "fr"]
+
+    date_received = fields.PCDateField("Date de réception de l'avis")
+
+    notice_type = fields.PCSelectField(
+        "Type d'avis",
+        choices=forms_utils.choices_from_enum(offerers_models.NoticeType, formatter=filters.format_notice_type),
+        validators=[DataRequired("Information obligatoire")],
+    )
+
+    amount = fields.PCDecimalField(
+        "Montant total dû à l'acteur culturel (en euros)",
+        use_locale=True,
+        validators=[DataRequired("Information obligatoire")],
+    )
+
+    reference = fields.PCStringField("Référence")
+    emitter_name = fields.PCStringField("Nom de l'émetteur")
+    emitter_email = fields.PCEmailField("Adresse mail de l'émetteur")
+
+    offerer = fields.PCTomSelectField(
+        "Entité juridique",
+        multiple=False,
+        choices=[],
+        validate_choice=False,
+        endpoint="backoffice_web.autocomplete_offerers",
+    )
+    venue = fields.PCTomSelectField(
+        "Structure (avec SIRET)",
+        multiple=False,
+        choices=[],
+        validate_choice=False,
+        endpoint="backoffice_web.autocomplete_pricing_points",
+    )
+
+    def validate(self, extra_validators: dict | None = None) -> bool:
+        offerer_id = self.offerer.data[0] if (self.offerer.data and self.offerer.data[0]) else None
+        venue_id = self.venue.data[0] if (self.venue.data and self.venue.data[0]) else None
+
+        if offerer_id and venue_id:
+            if int(offerer_id) != (
+                db.session.query(offerers_models.Venue)
+                .filter(offerers_models.Venue.id == venue_id)
+                .one()
+                .managingOffererId
+            ):
+                flash(
+                    "La structure doit être sur l'entité juridique sélectionnée",
+                    "warning",
+                )
+                return False
```

Comment:
Les validateurs Pydantic ne font pas directement des messages flash, mais agrègent des erreurs. Essaie plutôt : 
```suggestion
                raise validators.ValidationError("La structure doit être sur l'entité juridique sélectionnée")
```